src/wwwroot/services/auth.js

```diff
@@ -37,31 +37,56 @@
       isUrlAllowed: isUrlAllowed,
       getDefaultUrl: getDefaultUrl
     };
+    var loginSession = null;
+    
+    init();
+    return authService;
 
-    var decodedToken = null;
-    var encodedToken = null;
-    var temporarilyAuthed = false;
-
-    encodedToken = $window.localStorage.getItem('jwtToken');
-    if (encodedToken) {
-      decodeToken();
-      // verify expiration
+    function init() {
+      var encodedToken = $window.localStorage.getItem('jwtToken');
+      if (encodedToken) {
+        try {
+          loginSession = decodeLoginSession(encodedToken);
+        } catch {
+          logOut();
+          return;
+        }
+      }
+      // To ensure having relayLogin item in legacy sessions
```

Comment:
If I am not wrong, this code should be only executed if _encodedToken_ has value.